cpp/oneapi/dal/table/backend/csr_table_impl.hpp

```diff
@@ -0,0 +1,195 @@
+    }
+#endif
+
+    template <typename T>
+    void pull_column(const detail::default_host_policy& policy,
+                     array<T>& block,
+                     std::int64_t column_index,
+                     const range& rows) const {
+        static_assert("column_accessor doesn't supported for csr table");
+    }
+
+#ifdef ONEDAL_DATA_PARALLEL
+    template <typename T>
+    void pull_column(const detail::data_parallel_policy& policy,
+                     array<T>& block,
+                     std::int64_t column_index,
+                     const range& rows,
+                     sycl::usm::alloc alloc) const {
+        static_assert("column_accessor doesn't supported for csr table");
+    }
+#endif
+
+    template <typename T>
+    void pull_sparse_block(const detail::default_host_policy& policy,
+                           detail::sparse_block<T>& block,
+                           const range& rows) const {
+        csr_info origin_info{ meta_.get_data_type(0),
+                              layout_,
+                              row_count_,
+                              col_count_,
+                              row_indices_[row_count_] - row_indices_[0],
+                              csr_indexing_ };
+
+        // Overflow is checked here
+        check_block_row_range(rows);
+
+        block_info block_info{ rows.start_idx,
+                               rows.get_element_count(row_count_),
+                               detail::csr_indexing::one_based };
+
+        csr_pull_sparse_block(policy,
+                              origin_info,
+                              block_info,
+                              data_,
+                              column_indices_,
+                              row_indices_,
+                              block,
+                              alloc_kind::host);
+    }
+
+private:
+    static table_metadata create_csr_metadata(std::int64_t feature_count, data_type dtype) {
+        auto default_ftype =
+            detail::is_floating_point(dtype) ? feature_type::ratio : feature_type::ordinal;
+
+        auto dtypes = array<data_type>::full(feature_count, dtype);
+        auto ftypes = array<feature_type>::full(feature_count, default_ftype);
+        return table_metadata{ dtypes, ftypes };
+    }
+
+    void check_block_row_range(const range& rows) const {
+        const std::int64_t range_row_count = rows.get_element_count(row_count_);
+        detail::check_sum_overflow(rows.start_idx, range_row_count);
+        if (rows.start_idx + range_row_count > row_count_) {
+            throw range_error{ detail::error_messages::invalid_range_of_rows() };
+        }
+    }
+
+    table_metadata meta_;
+    array<byte_t> data_;
+    array<std::int64_t> column_indices_;
+    array<std::int64_t> row_indices_;
+    std::int64_t col_count_;
+    std::int64_t row_count_;
+    data_layout layout_;
+    detail::csr_indexing csr_indexing_;
+};
+
+} // namespace oneapi::dal::backend
```

Comment:
What about `row_accessor` and `column_accessor` support in csr table?

Reply:
It was not implemented now, but the method was created